### ISSUE TITLE
fix: Failed to obtain access credentials. Error: invalid_grant Bad Request Status code: 400

### ISSUE THREAD

**Issue by binemmanuel** [shorebirdtech/shorebird#2200]
Shorebird 1.1.9 • git@github.com:shorebirdtech/shorebird.git
Flutter 3.22.1 • revision 2f366bd7c9cdb59a1231fdb13c985be48f47149b
Engine • revision e5fdf87c8fa18e584204987de09e6d86f462c87c

**Description**

I tried to initialize a shorebird

``` bash
shorebird init --verbose
```

***Output***
``` bash
[HTTP] POST https://oauth2.googleapis.com/token
Failed to obtain access credentials. Error: invalid_grant Bad Request Status code: 400

You can also file an issue if you think this is a bug. Please include the following log file in your report:
/Users/username/Library/Application Support/shorebird/logs/1717518647889_shorebird.log
```

**Additional Context**
[1717518920367_shorebird.log](https://github.com/user-attachments/files/15567767/1717518920367_shorebird.log)

**Comment by eseidel** [comment]
We've seen 3 other reports of this: https://github.com/search?q=repo%3Ashorebirdtech%2Fshorebird+invalid_grant&type=issues

**Comment by binemmanuel** [comment]
> We've seen 3 other reports of this: https://github.com/search?q=repo%3Ashorebirdtech%2Fshorebird+invalid_grant&type=issues

My search wasn't good then

**Comment by eseidel** [comment]
Oh!  Sorry @binemmanuel that wasn't directed at you.  It was more me suggesting that we have something to fix here (rather than that you've done something wrong).  Thank you for the report!

**Comment by eseidel** [comment]
@felangel Could this be an invalid token of some form? Some problem in our JWT handling?  Or more likely network connectivity issues?

**Comment by binemmanuel** [comment]
I got you, it just not usual for me to create an issue when there are other issues like mine. 

I signed out and in then it worked just fine which means it has nothing to do with network connection. 
@eseidel

**Comment by eseidel** [comment]
Oh, interesting.  I wonder if this is the failure mode when using an older-style token?  @bryanoltman

**Comment by eseidel** [comment]
We could detect failures like this and recommend trying `shorebird logout`?

**Comment by erickzanardo** [comment]
> We could detect failures like this and recommend trying `shorebird logout`?

Implemented on #2244
